agentlightning/store/client_server.py

```diff
@@ -1538,10 +1432,54 @@ async def _request_json(
                 last_exc = net_exc
                 client_logger.info(f"Network/session issue will be retried. Retrying the request {method}: {path}")
                 if not await self._wait_until_healthy(session):
+                    # Check shared flag - if server is not online, silently ignore error
+                    if self._server_online_flag is not None:
+                        with self._server_online_flag.get_lock():
+                            is_online = bool(self._server_online_flag.value)
+                        if not is_online:
+                            client_logger.debug(
+                                f"Server is not online (shared flag). Silently ignoring {type(net_exc).__name__} for {method}: {path}"
+                            )
+                            # Silently ignore - return None to indicate failure was expected
+                            return None
                     break  # server is not healthy, do not retry
+            except asyncio.CancelledError as cancel_exc:
+                # Cancellation can occur during async operations, especially during shutdown
+                client_logger.debug(f"Request cancelled: {method}: {path}", exc_info=True)
+                # Check shared flag - if server is not online, silently ignore error
+                if self._server_online_flag is not None:
+                    with self._server_online_flag.get_lock():
+                        is_online = bool(self._server_online_flag.value)
+                    if not is_online:
+                        client_logger.debug(
+                            f"Server is not online (shared flag). Silently ignoring CancelledError for {method}: {path}"
+                        )
+                        # Silently ignore - return None to indicate failure was expected
+                        return None
```

Comment:
I'm not sure whether returning none here is appropriate. For some methods, returning none is fatal and might cause subsequent errors. I think it might be better to raise a ServerShutdownError here and handle that error in runner/agent.py.